lib/src/main/kotlin/at/bitfire/ical4android/validation/FixInvalidDayOffsetPreprocessor.kt

```diff
@@ -14,7 +14,9 @@ object FixInvalidDayOffsetPreprocessor : StreamPreprocessor() {
         // Examples:
         // TRIGGER:-P2DT
         // TRIGGER:-PT2D
-        "^(DURATION|TRIGGER):-?P((T-?\\d+D)|(-?\\d+DT))\$",
+        // REFRESH-INTERVAL;VALUE=DURATION:PT1D
+        "(?:^|(DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER);VALUE=)" +
```

Comment:
> What is the `|` at position 5 of the regex needed for? 

It ensures that the matched string starts with either 
- "DURATION;VALUE=" (or other property),  
- "DURATION:" or "TRIGGER:"

> Shouldn't it fail because xxDURATION is not in the list of processed properties?

No, we could solve that with another circumflex to denote line start, like this:

```
(?:^|(^(DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER));VALUE=)(DURATION|TRIGGER):-?P((T-?\d+D)|(-?\d+DT))$
```

But then we can just used this shorter and more pleasant to look at version instead:

```
^((DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER);VALUE=)|(DURATION|TRIGGER):-?P((T-?\d+D)|(-?\d+DT))$
```

*regex updated*

Reply:
> But then we can just use this shorter and more pleasant to look at version instead:
> 
> ```
> ^((DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER);VALUE=)|(DURATION|TRIGGER):-?P((T-?\d+D)|(-?\d+DT))$
> ```

I need to reverse on this one. The non-capturing group (`(?:X)`) is needed after all. It ensures "xxDURATION" or "FREE-BUSY" are ignored. So we should just stick with the bit more complex version I guess ... 

> ```
> (?:^|(^(DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER));VALUE=)(DURATION|TRIGGER):-?P((T-?\d+D)|(-?\d+DT))$
> ```

-> _regex updated_